packages/angular_devkit/build_optimizer/src/transforms/force-pure-call.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import * as ts from 'typescript';
+
+
+const pureFunctionComment = '@__PURE__';
+const callExprWhitelist = [
+  'makeDecorator',
+  'makePropDecorator',
+  'makeParamDecorator',
+  'makeMetadataCtor',
```

Comment:
instead of listing them here, would it not be better to to leave some kind of comment in the original source code which would mark the function as pure? This way as functions are discovered they can be added automatically not just by Angular team but by application developers?

Reply:
I see this is a hack for the 
- https://github.com/Microsoft/TypeScript/issues/17689
- https://github.com/Microsoft/TypeScript/issues/17606

But it sounds like from reading the Issue that the comment is there but it is in a wrong location?